src/clients/node/src/index.ts

```diff
@@ -1,4 +1,44 @@
-const binding: Binding = require('./client.node')
+function getBinding (): Binding {
+  const { arch, platform } = process
+
+  const archMap = {
+    "arm64": "aarch64",
+    "x64": "x86_64"
+  }
+
+  const platformMap = {
+    "linux": "linux",
+    "darwin": "macos"
+  }
+
+  if (! (arch in archMap)) {
+    throw new Error(`Unsupported arch: ${arch}`)
+  }
+
+  if (! (platform in platformMap)) {
+    throw new Error(`Unsupported platform: ${platform}`)
+  }
+
+  // Need a better way of doing this. We need to use our musl
```

Comment:
In the Java client, it is done by checking `/proc/self/map_files/` for the loaded libraries.
It works on any musl-based distro, and not only Alpine.

https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/clients/java/src/main/java/com/tigerbeetle/JNILoader.java#L55-L87

Reply:
Awesome thanks @batiati - will do the same logic here